### PR TITLE
Refactor: Improve Testability and Increase Test Coverage

### PR DESCRIPTION
**Pull request by d-fan**
### Objective
Increase unit test coverage and improve code testability through dependency injection and orchestration refactoring.

### Changes
1. **Dependency Injection:** Refactored `GmailClient`, `LLMClient`, and `StateManager` to accept all configuration via constructor arguments, removing global `settings` dependencies.
2. **Pipeline Refactoring:** Decoupled `main.py` by introducing a `Pipeline` class. This allows the core orchestration and "Hard Stop" logic to be tested in isolation with mocked clients.
3. **New Tests:**
   - Added `tests/test_main.py` for comprehensive pipeline orchestration testing.
   - Introduced property-based testing in `tests/test_state_manager_proptest.py` using `Hypothesis` to verify watermark consistency across diverse execution scenarios.
   - Enhanced edge case coverage for malformed LLM responses in `tests/test_llm_client.py`.
4. **Coverage Improvement:** Increased total project coverage from 51% to 71%.
5. **Quality:** All 31 tests pass and linting (`ruff`) is clean.

### Verification
- **Unit Tests:** `PYTHONPATH=src ./.venv/bin/python -m pytest tests/`
- **Coverage:** `PYTHONPATH=src ./.venv/bin/python -m pytest --cov=recruiter_auto_respond tests/`
- **Linting:** `./.venv/bin/python -m ruff check .`

### CODE REVIEW

File: tests/test_state_manager_proptest.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+from hypothesis import given
+from hypothesis import strategies as st
+
+from recruiter_auto_respond.state_manager import StateManager
+
+# Strategy for ISO timestamps
+iso_timestamps = st.datetimes().map(lambda dt: dt.strftime("%Y-%m-%dT%H:%M:%SZ"))
+
+# Strategy for results: list of (timestamp, success)
+results_strategy = st.lists(st.tuples(iso_timestamps, st.booleans()))
```
[the remark this answers is not in the document]
Comment (d-fan) [reply tests/test_state_manager_proptest.py:21]: Addressed. Constrained the `iso_timestamps` strategy to years 1900-2100 and added explicit UTC awareness to ensure reliable `strftime` behavior.

File: src/recruiter_auto_respond/llm_client.py [hunk 2]
```diff
@@ -56,32 +72,24 @@ def __init__(self, api_url: str, api_key: str = "sk-no-key-required") -> None:
...
             return {"Authorization": f"Basic {encoded_auth}"}
-
-        # Use provided api_key or fall back to settings
-        # In tests, sk-no-key-required is the default value passed to __init__
-        token = self.api_key
-        if not token or token == "sk-no-key-required":
-            token = getattr(settings, "LLM_API_KEY", "sk-no-key-required")
-
-        return {"Authorization": f"Bearer {token}"}
+        return {"Authorization": f"Bearer {self.api_key}"}
```
[the remark this answers is not in the document]
Comment (d-fan) [reply src/recruiter_auto_respond/llm_client.py:88]: Addressed. Updated `_get_headers()` to return an empty dictionary when credentials are not provided, preventing malformed `Bearer None` headers.

File: tests/test_state_manager_proptest.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import asyncio
+import os
+import tempfile
+
+from hypothesis import given
+from hypothesis import strategies as st
```
[the remark this answers is not in the document]
Comment (d-fan) [reply tests/test_state_manager_proptest.py:7]: Addressed. Added `hypothesis` and `pytest-cov` to the `dev` optional dependencies in `pyproject.toml`.

File: tests/test_main.py [hunk 4]
```diff
@@ -3,90 +3,140 @@
...
-                )
-                + 100  # Ensure it's after the watermark
-            ),
-        }
 
-        # 3. Mock Gmail: get_or_create_label
-        mock_gmail.get_or_create_label.return_value = "label_id"
+@pytest.fixture
+def mock_state_manager():
+    return AsyncMock()
```
[the remark this answers is not in the document]
Comment (d-fan) [reply tests/test_main.py:28]: Addressed. Annotated all fixtures with concrete return types (e.g., `AsyncMock`) to satisfy the repo's `strict` type-checking configuration.

File: tests/test_main.py [hunk 5]
```diff
@@ -3,90 +3,140 @@
...
-        ), patch(
-            "recruiter_auto_respond.main.setup_clients",
-            return_value=(mock_gmail, mock_sheets, mock_llm),
-        ), patch(
-            "argparse.ArgumentParser.parse_args",
-            return_value=MagicMock(dry_run=False),
-        ):
-            await main()
+@pytest.mark.asyncio
+async def test_pipeline_run_no_messages(
```
[the remark this answers is not in the document]
Comment (d-fan) [reply tests/test_main.py:51]: Addressed. Added explicit return types and parameter types to all test functions.

File: tests/test_state_manager_proptest.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    async def run_test():
+        manager = StateManager(temp_state_file)
+        await manager.save_state({"last_run_timestamp": initial_ts})
+
+        # Sort results by timestamp as the actual pipeline would
+        sorted_results = sorted(results, key=lambda x: x[0])
+
+        new_ts = await manager.update_watermark(sorted_results)
+
```
[the remark this answers is not in the document]
Comment (d-fan) [reply tests/test_state_manager_proptest.py:37]: Addressed. Added explicit type annotations to all generated test inputs and the nested coroutine to satisfy `strict` typing requirements.

File: src/recruiter_auto_respond/main.py [hunk 7]
```diff
@@ -13,210 +13,275 @@
...
+                        f"[DRY-RUN] Would label message {message_id} as Recruiter."
+                    )
+
+                # Prepare row for Sheets
+                row = {
+                    "threadId": thread_id,
+                    "messageId": message_id,
+                    "timestamp": msg_ts_iso,
+                }
             else:
```
[the remark this answers is not in the document]
Comment (d-fan) [reply src/recruiter_auto_respond/main.py:79]: Deferred to Issue #28. While `_classify_and_record` checks the `stop_event`, a strict hard stop across concurrent tasks requires moving to a worker-pool pattern. This has been documented for future implementation to maintain sync reliability without excessive code complexity in this PR.
